account/lnurl.go

```diff
@@ -62,6 +62,25 @@ func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err
 
 	rawurl, iparams, err := lnurl.HandleLNURL(encodedLnurl)
 	if err != nil {
+
+		if lightningAddress != "" {
```

Comment:
It seems weird to have two different types of errors (one for lightning address and one for lnurl).
Why not using the data.LNUrlREsponse with error field set to all cases?

Reply:
Agreed. But I've reverted the addition of LNUrlResponseError in the latest commits. I don't think I need it, after all, to handle the errors properly. Hopefully the new method is better.